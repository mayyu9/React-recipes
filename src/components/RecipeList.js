import React from 'react';
import PropTypes from 'prop-types';
import RecipeListItem from './RecipeListItem';

const RecipeList = ({ recipes, style, ...props }) => {
  return (
    <div style={style}>
      <h2 className="h2">Recipes</h2>
      <ul className="list-reset">
        {recipes.map(recipe => (
          <RecipeListItem
            key={recipe.id}
            recipe={recipe}
            {...props}
          />
        ))}
      </ul>
    </div>
  );
};

RecipeList.propTypes = {
  style: PropTypes.object,
  recipes: PropTypes.array,
  onClick: PropTypes.func,
  onFavourited: PropTypes.func,
};

RecipeList.defaultProps = {
  recipes: [],
};

export default RecipeList;
