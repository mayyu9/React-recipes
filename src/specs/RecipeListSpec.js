import React from 'react';
import renderer from 'react-test-renderer';
import RecipeList from '../components/RecipeList';

const testRecipe = [
  {
    id: 1,
    name: 'test Recipe',
    category: 'test category',
  },
];

describe('<RecipeList />', () => {
  test('should not break if no recipe passed to RecipeList component', () => {
    const component = renderer.create(
      <RecipeList />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('should render when recipe passed to RecipeListItem component', () => {
    const component = renderer.create(
      <RecipeList recipes={testRecipe} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
