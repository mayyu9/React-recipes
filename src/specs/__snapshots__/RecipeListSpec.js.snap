// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeList /> should not break if no recipe passed to RecipeList component 1`] = `
<div>
  <h2
    className="h2"
  >
    Recipes
  </h2>
  <ul
    className="list-reset"
  />
</div>
`;

exports[`<RecipeList /> should render when recipe passed to RecipeListItem component 1`] = `
<div>
  <h2
    className="h2"
  >
    Recipes
  </h2>
  <ul
    className="list-reset"
  >
    <li
      className="py2 border-bottom border-bottom-dashed pointer"
      onClick={[Function]}
    >
      <span
        className="mr1"
      >
        <input
          onClick={[Function]}
          type="checkbox"
        />
      </span>
      <span>
        test Recipe
      </span>
      <span>
        test category
      </span>
    </li>
  </ul>
</div>
`;
